pkg/R/utils.R

```diff
@@ -419,3 +419,82 @@ cleanClosure <- function(func, checkedFuncs = new.env()) {
   }
   func
 }
+
+# Common function for implmentation of zipRDD() and cartesian().
+# param
+#   x An RDD.
+#   other An RDD.
+#   operation The specified operation, "zip" or "cartesian".
+#
+# return value
+#   The result RDD.
+doZipOrCartesian <- function(x, other, operation = c("zip", "cartesian")) {
```

Comment:
I did think about split the single common function into two common functions as mentioned above. But I felt the logic of the two common functions are related and the smell does not feel so good. Anyway, I put a WIP at https://github.com/amplab-extras/SparkR-pkg/commit/02363086977cbbbd2c159c81dac5d2e9617263b4 for this refactoring for your review. If you prefer this way, I will make the change here.